src/components/Register/Register.js

```diff
@@ -0,0 +1,119 @@
+import "../PopupWithForm/__error-message/_bad-email/popup-with-form__error-message_bad-email.css";
+import "../PopupWithForm/__send-button/popup-with-form__send-button.css";
+import "../PopupWithForm/__send-button/_disabled/popup-with-form__send-button_disabled.css";
+import "../PopupWithForm/__or-block/popup-with-form__or-block.css";
+import "../PopupWithForm/__or/popup-with-form__or.css";
+import "../PopupWithForm/__or-button/popup-with-form__or-button.css";
+
+import Popup from "../Popup/Popup.js";
+
+function PopupWithForm({
+  isOpen,
+  onClose,
+  onRegister,
+  handleChange,
+  values,
+  errors,
+  isValid,
+  isEmailAvailable,
+  moveToSignInForm,
+}) {
+  return (
+    <Popup isOpen={isOpen} onClose={onClose}>
+      <form className="popup-with-form">
+        <h2 className="popup-with-form__title">Sign up</h2>
+
+        <label className="popup-with-form__label">Email</label>
+        <input
+          type="email"
+          name="email"
+          id="email-input"
+          placeholder="Enter email"
+          className="popup-with-form__input"
+          noValidate
+          required
+          onChange={handleChange}
+          value={values.email || ""}
+        />
+        <div className="popup-with-form__error-message">
+          {`${errors.email ? errors.email : ""}`}
+        </div>
+
+        <label className="popup-with-form__label">Password</label>
+        <input
+          type="password"
+          name="password"
+          id="password-input"
+          placeholder="Enter password"
+          className="popup-with-form__input"
+          noValidate
+          required
+          onChange={handleChange}
+          value={values.password || ""}
+          minLength="8"
+          maxLength="30"
+        />
+        <div className="popup-with-form__error-message">
+          {`${errors.password ? errors.password : ""}`}
+        </div>
+
+        <label className="popup-with-form__label">Username</label>
+        <input
+          type="text"
+          name="username"
+          id="username-input"
+          placeholder="Enter your username"
+          className="popup-with-form__input"
+          minLength="2"
+          maxLength="30"
+          noValidate
+          required
+          onChange={handleChange}
+          value={values.username || ""}
+        />
+        <div className="popup-with-form__error-message">
+          {`${errors.username ? errors.username : ""}`}
+        </div>
+
+        <div
+          className="popup-with-form__error-message popup-with-form__error-message_bad-email"
+          style={{
+            display: `${!isEmailAvailable ? "" : "none"}`,
+          }}
+        >
+          This email is not available
+        </div>
+        <button
+          type="submit"
+          className={`popup-with-form__send-button ${
+            !isValid ? "popup-with-form__send-button_disabled" : ""
+          }`}
+          onClick={(event) => {
```

Comment:
Submit handlers are added only to tags `form` with event `submit` rather than to submit buttons with event `click` because it also automatically handles `Enter` presses